src/git_analyzer.sh

```
@@ -0,0 +1,169 @@
+            change_type="deleted"
+        elif git diff --cached --name-status | grep -q "^R.*${filename}$"; then
+            change_type="renamed"
+        fi
+        
+        # JSONエントリを構築（数値の安全な処理）
+        local safe_additions="${additions//[^0-9]/}"
+        local safe_deletions="${deletions//[^0-9]/}"
+        safe_additions="${safe_additions:-0}"
+        safe_deletions="${safe_deletions:-0}"
+        
+        local file_entry=$(cat <<EOF
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)